File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/fanmanagement/FanManagementAction.kt
package com.hhp227.concafe.presentation.main.fanmanagement

sealed interface FanManagementAction {
    data object ClickClaimProfile : FanManagementAction
    data object LoadMoreClaimCandidates : FanManagementAction
    data class SelectClaimCandidate(val castId: String) : FanManagementAction
    data object SubmitCastClaim : FanManagementAction
    data object DismissClaimSheet : FanManagementAction
    data object ClickEditProfile : FanManagementAction
    data object ClickPrimaryAnnouncement : FanManagementAction
    data class ChangeAnnouncementTitle(val value: String) : FanManagementAction
    data class ChangeAnnouncementBody(val value: String) : FanManagementAction
    data object SubmitAnnouncement : FanManagementAction
    data object DismissAnnouncementSheet : FanManagementAction
    data class ClickQuickAction(val quickAction: FanManagementUiState.QuickAction) : FanManagementAction
    data object ClickViewAllFollowers : FanManagementAction
    data class ClickRecentFollower(val followerId: String) : FanManagementAction
    data class ClickTopFan(val fanId: String) : FanManagementAction
    data object DismissInfoMessage : FanManagementAction
}

File: iosApp/iosApp/Presentation/Main/FanManagement/FanManagementAction.swift
//
//  FanManagementAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation

enum FanManagementAction {
    case clickClaimProfile
    case loadMoreClaimCandidates
    case selectClaimCandidate(String)
    case submitCastClaim
    case dismissClaimSheet
    case clickEditProfile
    case clickPrimaryAnnouncement
    case changeAnnouncementTitle(String)
    case changeAnnouncementBody(String)
    case submitAnnouncement
    case dismissAnnouncementSheet
    case clickQuickAction(FanManagementUiState.QuickAction)
    case clickViewAllFollowers
    case clickRecentFollower(id: String)
    case clickTopFan(id: String)
    case dismissInfoMessage
}
